### PR TITLE
Feature/add lognsequence sample

### PR DESCRIPTION
**Pull request by pablogs9**
This PR adds two sample applications where a long char array is published in order to test fragmentation.

### CODE REVIEW

File: rcl/long_sequence_publisher/main.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  std_msgs__msg__String msg;
+  msg.data.data = (char * ) malloc(ARRAY_LEN * sizeof(char));
+  msg.data.size = 0;
+  msg.data.capacity = ARRAY_LEN;
+
+  // Fill the array with a known sequence
+  for (size_t i = 0; i < 3500; i++){
+    msg.data.data[i] = (char) 'z';
+    msg.data.size++;
+  }
```
Comment (julibert) [inline rcl/long_sequence_publisher/main.c:41]: You can use [memset](https://en.cppreference.com/w/c/string/byte/memset) here.

File: rcl/long_sequence_subscriber/main.c [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+      rcl_ret_t rc = rcl_take(wait_set.subscriptions[index], &msg, NULL, NULL);
+      if (RCL_RET_OK == rc) {
+        
+        bool pass_test = true;
+        // Check if sequence items matches the know pattern
+        for (size_t i = 0; i <  msg.data.size; i++){
+          if (msg.data.data[i] != 'z'){
+            pass_test = false;
+          }
```
Comment (julibert) [inline rcl/long_sequence_subscriber/main.c:58]: You could use [strcmp](https://en.cppreference.com/w/c/string/byte/strcmp) here.

File: rcl/long_sequence_publisher/main.c (hunk 1 above)
Comment (pablogs9) [reply rcl/long_sequence_publisher/main.c]: Done

File: rcl/long_sequence_subscriber/main.c (hunk 2 above)
Comment (pablogs9) [reply rcl/long_sequence_subscriber/main.c]: Done

File: rcl/CMakeLists.txt [hunk 3]
```diff
@@ -41,6 +41,8 @@ export_executable(int32_multinode)
 export_executable(guard_condition)
 export_executable(int32_publisher_subscriber)
 export_executable(ping_pong)
+export_executable(long_sequence_publisher)
```
Comment (julibert) [inline rcl/CMakeLists.txt:44]: The name could be confused with "a sequence of longs" which is something completely different.
I would use something like "fragmented_publication" and "fragmented_subscription".

Comment (pablogs9) [reply rcl/CMakeLists.txt]: Done, merge if looks good to you

**Pull request merged**